### PR TITLE
Add check for shared mem payload type

### PR DESCRIPTION
**Pull request by brianeaton**
Any messages received which have UPayloadFormat of UPAYLOAD_FORMAT_SHM should have their UAttributes logged and then dropped.

### CODE REVIEW

File: up-streamer/src/ustreamer.rs [hunk 1]
```diff
@@ -712,6 +712,18 @@ impl UListener for ForwardingListener {
             FORWARDING_LISTENER_FN_ON_RECEIVE_TAG,
             &msg
         );
+
+        if msg.attributes.payload_format.enum_value_or_default() == up_rust::UPayloadFormat::UPAYLOAD_FORMAT_SHM {
+            debug!(
+                "{}:{}:{} Received message with type UPAYLOAD_FORMAT_SHM, which is not supported: {:#?}",
```
Comment (PLeVasseur) [inline up-streamer/src/ustreamer.rs:718]:
```suggestion
                "{}:{}:{} Received message with type UPAYLOAD_FORMAT_SHM, which is not supported. A pointer to shared memory will not be usable on another device. UAttributes: {:#?}",
```

**Review by PLeVasseur — CHANGES_REQUESTED** [review]
Hey @brianeaton -- thanks for the PR!

Take a look at some suggestions below.

**Comment by PLeVasseur** [comment]
I think I need to fix the workflow triggers for this repo, as they are not firing. I'll go do so, then ask you to rebase and we'll make sure the workflow triggers.

**Comment by PLeVasseur** [comment]
Workflow triggers fix has been merged. Feel free to now rebase and squash to a meaningful commit history.

File: up-streamer/src/ustreamer.rs [hunk 2]
```diff
@@ -712,6 +712,18 @@ impl UListener for ForwardingListener {
             FORWARDING_LISTENER_FN_ON_RECEIVE_TAG,
             &msg
         );
+
+        if msg.attributes.payload_format.enum_value_or_default() == UPayloadFormat::UPAYLOAD_FORMAT_SHM {
```
Comment (matthewd0123) [inline up-streamer/src/ustreamer.rs:716]: Rust convention is maximum line length of 100 characters: https://doc.rust-lang.org/nightly/style-guide/

```suggestion
        if msg.attributes.payload_format.enum_value_or_default()
            == UPayloadFormat::UPAYLOAD_FORMAT_SHM
        {
            debug!(
                "{}:{}:{} Received message with type UPAYLOAD_FORMAT_SHM, \
                which is not supported. A pointer to shared memory will not \
                be usable on another device. UAttributes: {:#?}",
                self.forwarding_id,
                FORWARDING_LISTENER_TAG,
                FORWARDING_LISTENER_FN_ON_RECEIVE_TAG,
                &msg.attributes
            );
            return;
        }
```

**Review by matthewd0123 — APPROVED** [review]
Looks good to me! Just one comment on formatting.

File: up-streamer/src/ustreamer.rs (hunk 2 above)
Comment (brianeaton) [reply up-streamer/src/ustreamer.rs:716]: Thanks for pointing it out. I just resolved that in my latest commit.

**Review by PLeVasseur — APPROVED** [review]
LGTM

Can you rebase now that #20 has merged?

**Pull request merged**